packages/mui-joy/src/styles/extendTheme.ts

```diff
@@ -243,145 +279,166 @@ export default function extendTheme(themeOptions?: CssVarsThemeOptions): Theme {
-        solidColor: getCssVar(`palette-common-black`),
-        solidBg: getCssVar(`palette-warning-300`),
-        solidHoverBg: getCssVar(`palette-warning-400`),
-        solidActiveBg: getCssVar(`palette-warning-500`),
+        solidColor: getCssVarColor(`palette-common-black`),
+        solidBg: getCssVarColor(`palette-warning-300`),
+        solidHoverBg: getCssVarColor(`palette-warning-400`),
+        solidActiveBg: getCssVarColor(`palette-warning-500`),
       },
       common: {
         white: '#FFF',
         black: '#09090D',
       },
       text: {
-        primary: getCssVar('palette-neutral-100'),
-        secondary: getCssVar('palette-neutral-300'),
-        tertiary: getCssVar('palette-neutral-400'),
+        primary: getCssVarColor('palette-neutral-100'),
+        secondary: getCssVarColor('palette-neutral-300'),
+        tertiary: getCssVarColor('palette-neutral-400'),
       },
       background: {
-        body: getCssVar('palette-neutral-900'),
-        surface: getCssVar('palette-common-black'),
-        popup: getCssVar('palette-neutral-800'),
-        level1: getCssVar('palette-neutral-800'),
-        level2: getCssVar('palette-neutral-700'),
-        level3: getCssVar('palette-neutral-600'),
-        tooltip: getCssVar('palette-neutral-600'),
-        backdrop: `rgba(${getCssVar('palette-neutral-darkChannel')} / 0.5)`,
-      },
-      divider: `rgba(${getCssVar('palette-neutral-mainChannel')} / 0.24)`,
-      focusVisible: getCssVar('palette-primary-500'),
+        body: getCssVarColor('palette-neutral-900'),
+        surface: getCssVarColor('palette-common-black'),
+        popup: getCssVarColor('palette-neutral-800'),
+        level1: getCssVarColor('palette-neutral-800'),
+        level2: getCssVarColor('palette-neutral-700'),
+        level3: getCssVarColor('palette-neutral-600'),
+        tooltip: getCssVarColor('palette-neutral-600'),
+        backdrop: `rgba(${getCssVarColor('palette-neutral-darkChannel')} / 0.5)`,
+      },
+      divider: `rgba(${getCssVar(
+        'palette-neutral-mainChannel',
+        colorChannel(defaultColors.neutral[500]),
+      )} / 0.24)`,
+      focusVisible: getCssVarColor('palette-primary-500'),
     },
     shadowRing: '0 0 #000',
     shadowChannel: '0 0 0',
   };
 
+  const fontFamilyFallback =
+    '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"';
+  const fontFamily = {
+    body: `"Public Sans", ${getCssVar('fontFamily-fallback', fontFamilyFallback)}`,
+    display: `"Public Sans", ${getCssVar('fontFamily-fallback', fontFamilyFallback)}`,
+    code: 'Source Code Pro,ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,Liberation Mono,Courier New,monospace',
+    fallback: fontFamilyFallback,
+    ...scalesInput.fontFamily,
+  };
+
+  const fontWeight = {
+    xs: 200,
+    sm: 300,
+    md: 500,
+    lg: 600,
+    xl: 700,
+    xl2: 800,
+    xl3: 900,
+    ...scalesInput.fontWeight,
```

Comment:
These values need to be merged earlier, so that the CSS var default values reflect the correct CSS value.